nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/cf/JndiJmsConnectionFactoryProvider.java

```diff
@@ -21,214 +21,55 @@
-        .displayName("Initial Naming Factory Class")
-        .description("The fully qualified class name of the Java Initial Naming Factory (java.naming.factory.initial).")
-        .addValidator(NON_EMPTY_VALIDATOR)
-        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
-        .required(true)
-        .build();
-    static final PropertyDescriptor NAMING_PROVIDER_URL = new Builder()
-        .name("java.naming.provider.url")
-        .displayName("Naming Provider URL")
-        .description("The URL of the JNDI Naming Provider to use")
-        .required(true)
-        .addValidator(NON_EMPTY_VALIDATOR)
-        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
-        .build();
-    static final PropertyDescriptor CONNECTION_FACTORY_NAME = new Builder()
-        .name("connection.factory.name")
-        .displayName("Connection Factory Name")
-        .description("The name of the JNDI Object to lookup for the Connection Factory")
-        .required(true)
-        .addValidator(NON_EMPTY_VALIDATOR)
-        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
-        .build();
-    static final PropertyDescriptor NAMING_FACTORY_LIBRARIES = new Builder()
-        .name("naming.factory.libraries")
-        .displayName("Naming Factory Libraries")
-        .description("Specifies .jar files or directories to add to the ClassPath in order to find the Initial Naming Factory Class")
-        .required(false)
-        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
-        .addValidator(StandardValidators.createListValidator(true, true, StandardValidators.createURLorFileValidator()))
-        .dynamicallyModifiesClasspath(true)
-        .build();
-    static final PropertyDescriptor PRINCIPAL = new Builder()
-        .name("java.naming.security.principal")
-        .displayName("JNDI Principal")
-        .description("The Principal to use when authenticating with JNDI")
-        .required(false)
-        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
-        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
-        .build();
-    static final PropertyDescriptor CREDENTIALS = new Builder()
-        .name("java.naming.security.credentials")
-        .displayName("Credentials")
-        .description("The Credentials to use when authenticating with JNDI")
-        .required(false)
-        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
-        .addValidator(Validator.VALID)
-        .sensitive(true)
-        .build();
-
-    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Arrays.asList(
-        INITIAL_NAMING_FACTORY_CLASS,
-        NAMING_PROVIDER_URL,
-        CONNECTION_FACTORY_NAME,
-        NAMING_FACTORY_LIBRARIES,
-        PRINCIPAL,
-        CREDENTIALS);
-
-    private ConnectionFactory connectionFactory;
+    private JndiJmsConnectionFactoryHandler delegate;
```

Comment:
This needs to be protected also in terms of concurrency. Looks like marking as `volatile` will be sufficient in this case.